### PR TITLE
Feature add popup messages for create and update users and reports

### PR DESCRIPTION
**Pull request by marsolekk**

### CODE REVIEW

File: employees/static/employees/scripts/expandReportList.js [hunk 1]
```diff
@@ -1,10 +1,18 @@
 window.expandReportList = function (element, numberOfReports){
-    var currentRow = element.parentElement.parentElement
+    var currentRow = element.parentElement.parentElement;
     var nextTableRow = currentRow.nextElementSibling;
 
     var tipIcon = element.getElementsByClassName("rotated-tip")[0];
     var verticalLineIcon = element.getElementsByClassName("report-list-vl")[0];
 
+    let expandReportTitle = "Expand report list"
+    if (element.getAttribute("title") === expandReportTitle){
```
Comment (kbeker) [inline employees/static/employees/scripts/expandReportList.js:9]: Missing space between `)` and `{`

File: employees/static/employees/scripts/expandReportList.js [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+        element.setAttribute("title", "Roll report list")
+    }
+    else{
```
Comment (kbeker) [inline employees/static/employees/scripts/expandReportList.js:12]: Missing space between `else` and `{`

File: employees/templates/employees/partial/popup_message.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{% for message in messages %}
```
Comment (kbeker) [inline employees/templates/employees/partial/popup_message.html:1]: Why two identical partial templates? Why not used common place to place one of them? Or am I blind and they are different? :p

File: users/templates/partials/popup_message.html [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{% for message in messages %}
```
Comment (kbeker) [inline users/templates/partials/popup_message.html:1]: Why two identical partial templates? Why not used common place to place one of them? Or am I blind and they are different? :p

File: users/views.py [hunk 5]
```diff
@@ -98,6 +100,9 @@ def get_context_data(self, *, _object_list: Any = None, **kwargs: Any) -> dict:
     def get_success_url(self) -> str:
         return reverse("custom-users-list")
 
+    def get_success_message(self, cleaned_data: Any) -> str:
+        return self.success_message.format(email=self.object.email, action="created")
```
Comment (kbeker) [inline users/views.py:104]: You can create separate enum for actions and move them to constants or so

File: users/views.py [hunk 6]
```diff
@@ -156,6 +162,9 @@ def get_context_data(self, *, _object_list: Any = None, **kwargs: Any) -> dict:
         context_data["is_create_user_form"] = False
         return context_data
 
+    def get_success_message(self, cleaned_data: Any) -> str:
+        return self.success_message.format(email=self.object.email, action="updated")
```
Comment (kbeker) [inline users/views.py:166]: Yea, please move actions to enum in constants

File: employees/templates/employees/author_report_list.html [hunk 7]
```diff
@@ -46,4 +47,9 @@
         integrity="{% staticinline 'employees/scripts/modalsFunctionality.js' encode="sri" %}"
         crossorigin="anonymous">
     </script>
+    <script
+        src="{% static 'users/scripts/hidePopupMessage.js' %}"
+        integrity="{% staticinline 'users/scripts/hidePopupMessage.js' encode="sri" %}"
```
Comment (kbeker) [inline employees/templates/employees/author_report_list.html:52]: Are you sure you did here write quotes? First after equal sign is `"` then again you're using it after next equal sign with encode parameter. Shouldn't be there single quote sign `'`?

File: employees/tests/test_unit_custom_forms.py [hunk 8]
```diff
@@ -9,28 +9,37 @@
...
-        for i in range(queryset_length):
-            self.assertEqual(choices[i][0], queryset[i].id)
-            self.assertEqual(choices[i][1], queryset[i].name)
+        self.assertEqual(len(choices), self.queryset_length)
+
+        self._compare_id_and_names_of_choices_and_project_for_projects(choices)
+
+    @staticmethod
+    def _generate_test_projects(queryset_length: int):
+        for _project_number in range(queryset_length):
```
Comment (kbeker) [inline employees/tests/test_unit_custom_forms.py:36]: You can just use here `_` instead of `_project_number`

**Review by kbeker — COMMENTED** [review]
Few fixes to be done. Let me know once you fix them

File: employees/static/employees/scripts/expandReportList.js (hunk 2 above)
Comment (marsolekk) [reply employees/static/employees/scripts/expandReportList.js:12]: Done

File: employees/static/employees/scripts/expandReportList.js (hunk 1 above)
Comment (marsolekk) [reply employees/static/employees/scripts/expandReportList.js:9]: Done

File: employees/tests/test_unit_custom_forms.py (hunk 8 above)
Comment (marsolekk) [reply employees/tests/test_unit_custom_forms.py:36]: Done

File: employees/templates/employees/author_report_list.html (hunk 7 above)
Comment (marsolekk) [reply employees/templates/employees/author_report_list.html:52]: More proper would be a single quote sign `' ` for sure. I changed that. However, I believe that as it is surrounded by `{%` and `%}` it doesn't affect the correct work.

File: users/templates/partials/popup_message.html (hunk 4 above)
Comment (marsolekk) [reply users/templates/partials/popup_message.html]: Done. Hope now it is proper solution. I moved that to `users/templates/common/`

File: users/views.py (hunk 6 above)
Comment (marsolekk) [reply users/views.py]: I’m not sure that I’ve done what you requested, so please let me know, so I will change it if needed.

**Review by kbeker — APPROVED** [review]
Looks good, ready to merge

**Pull request merged**